pkg/controllers/routing/utils.go

```diff
@@ -120,6 +120,26 @@ func getNodeSubnet(nodeIP net.IP) (net.IPNet, string, error) {
 	return net.IPNet{}, "", errors.New("Failed to find interface with specified node ip")
 }
 
+func getMTUFromNodeIP(nodeIP net.IP) (int, error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return 0, errors.New("Failed to get list of links")
+	}
+	for _, link := range links {
+		addresses, err := netlink.AddrList(link, netlink.FAMILY_ALL)
+		if err != nil {
+			return 0, errors.New("Failed to get list of addr")
+		}
+		for _, addr := range addresses {
+			if addr.IPNet.IP.Equal(nodeIP) {
+				lintMTU := link.Attrs().MTU
+				return lintMTU - 20, nil // -20 to accommodate IPIP header
```

Comment:
You should do testing here to see if the user is running an overlay before automatically reducing the linkMTU by 20 bytes. We don't run our nodes in an overlay network and the 20 byte difference here a .26 Gbps difference in our performance testing when testing container to container where jumbo frames were enabled on both nodes.

In my testing, I just passed a boolean to `getMTUFromNodeIP` that I sourced from `nrc.enableOverlays`. I don't know if `nrc.enableOverlays` takes into account all possible IPIP overlay scenarios or not without going over `injectRoute` in more fine grained detail.

Reply:
My mocked up `getMTUFromNodeIP` looked like:
```
func getMTUFromNodeIP(nodeIP net.IP, overlayEnabled bool) (int, error) {
	links, err := netlink.LinkList()
	if err != nil {
		return 0, errors.New("Failed to get list of links")
	}
	for _, link := range links {
		addresses, err := netlink.AddrList(link, netlink.FAMILY_ALL)
		if err != nil {
			return 0, errors.New("Failed to get list of addr")
		}
		for _, addr := range addresses {
			if addr.IPNet.IP.Equal(nodeIP) {
				lintMTU := link.Attrs().MTU
				if overlayEnabled {
					return lintMTU - 20, nil // -20 to accommodate IPIP header
				}
				return lintMTU, nil
			}
		}
	}
	return 0, errors.New("Failed to find interface with specified node ip")
}
```

And then on the calling side, I had something like:
```
func (nrc *NetworkRoutingController) autoConfigureMTU() error {
	mtu, err := getMTUFromNodeIP(nrc.nodeIP, nrc.enableOverlays)
	if err != nil {
		return fmt.Errorf("failed to generate MTU: %s", err.Error())
	}
```